src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.openloadflow.graph.GraphConnectivity;
+import com.powsybl.security.action.Action;
+import com.powsybl.security.action.SwitchAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class LfAction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LfAction.class);
+
+    private final String id;
+
+    private LfBranch disabledBranch; // switch to open
+
+    private LfBranch enabledBranch; // switch to close
+
+    private LfNetwork network;
+
+    public LfAction(Action action, LfNetwork network) {
+        this.id = Objects.requireNonNull(action.getId());
+        this.network = Objects.requireNonNull(network);
+        switch (action.getType()) {
+            case SwitchAction.NAME:
+                SwitchAction switchAction = (SwitchAction) action;
+                LfBranch branch = network.getBranchById(switchAction.getSwitchId());
+                if (branch == null) {
+                    throw new PowsyblException("Branch " + switchAction.getSwitchId() + " not found in the network");
+                }
+                if (switchAction.isOpen()) {
+                    disabledBranch = branch;
+                } else {
+                    enabledBranch = branch;
+                }
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported action type: "  + action.getType());
+        }
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public LfBranch getDisabledBranch() {
+        return disabledBranch;
+    }
+
+    public LfBranch getEnabledBranch() {
+        return enabledBranch;
+    }
+
+    public void apply() {
+        apply(false, Collections.emptyList());
+    }
+
+    public void apply(boolean withConnectivity, List<String> alreadyRemovedEdges) {
+        if (disabledBranch != null) {
+            disabledBranch.setDisabled(true);
+        }
+        if (enabledBranch != null) {
+            enabledBranch.setDisabled(false);
+            enabledBranch.getBus1().setDisabled(false);
```

Comment:
Do we need to enable buses ? In which case?

Reply:
Yes we need it (see the unit tests) and indeed in it really not enough. For example, when we reconnect a switch that will reconnect a small component, the buses and branches of this component but be enabled. This is not handle here. @flo-dup  how to do this with your connectivity?

Reply:
So yes, this is not the right thing to do. We should be able to get from connectivity, the list of buses and branches that have been reconnected to main connected component. Here by just enabling side buses, we only enable a small part of potential sub network reconnected.